src/index.ts

```diff
@@ -1,6 +1,7 @@
 import handlebars from "handlebars";
 import { PluginBuild, OnLoadOptions } from "esbuild";
 import { stat, readFile } from "fs/promises";
+import { minify } from "html-minifier-terser";
```

Comment:
I don't see this module in package.json, I think we should put it in peerDependencies and let people to install it when needed

In consequences, the minier module should be imported dynamically based on minify options, see next comment 

please add a little test of runtime check or both ESM and CJS contexts